authenticatorbridge/src/test/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerTest.kt

```diff
@@ -0,0 +1,440 @@
+        unmockkStatic(IAuthenticatorBridgeService.Stub::class)
+        unmockkStatic(EncryptedSharedAccountData::decrypt)
+    }
+
+    @Test
+    fun `initial state should be Loading`() {
+        assertEquals(AccountSyncState.Loading, manager.accountSyncStateFlow.value)
+    }
+
+    @Test
+    fun `initial state should be AppNotInstalled when package manager can't find app`() {
+        every {
+            context.packageManager.getPackageInfo("com.x8bit.bitwarden.dev", 0)
+        } throws NameNotFoundException()
+        val manager = AuthenticatorBridgeManagerImpl(
+            context = context,
+            connectionType = AuthenticatorBridgeConnectionType.DEV,
+            symmetricKeyStorageProvider = fakeSymmetricKeyStorageProvider,
+            callbackProvider = testAuthenticatorBridgeCallbackProvider,
+            processLifecycleOwner = fakeLifecycleOwner,
+        )
+        assertEquals(AccountSyncState.AppNotInstalled, manager.accountSyncStateFlow.value)
+    }
+
+    @Test
+    fun `onStart when bindService fails should set state to error`() {
+        val mockIntent: Intent = mockk()
+        every {
+            anyConstructed<Intent>().setComponent(any())
+        } returns mockIntent
+
+        every { context.bindService(any(), any(), Context.BIND_AUTO_CREATE) } returns false
+
+        fakeLifecycleOwner.lifecycle.dispatchOnStart()
+
+        assertEquals(AccountSyncState.Error, manager.accountSyncStateFlow.value)
+        verify { context.bindService(any(), any(), Context.BIND_AUTO_CREATE) }
+    }
+
+    @Test
+    fun `onStart when package manager doesn't find bitwarden app should set state to AppNotInstalled`() {
+        val mockIntent: Intent = mockk()
+        every {
+            context.packageManager.getPackageInfo("com.x8bit.bitwarden.dev", 0)
+        } throws NameNotFoundException()
+        every {
+            anyConstructed<Intent>().setComponent(any())
+        } returns mockIntent
+
+        every {
+            context.bindService(any(), any(), Context.BIND_AUTO_CREATE)
+        } throws SecurityException()
+
+        fakeLifecycleOwner.lifecycle.dispatchOnStart()
+
+        assertEquals(AccountSyncState.AppNotInstalled, manager.accountSyncStateFlow.value)
+    }
+
+    @Test
+    fun `onStart when bindService throws security exception should set state to error`() {
+        val mockIntent: Intent = mockk()
+        every {
+            anyConstructed<Intent>().setComponent(any())
+        } returns mockIntent
+
+        every {
+            context.bindService(any(), any(), Context.BIND_AUTO_CREATE)
+        } throws SecurityException()
+
+        fakeLifecycleOwner.lifecycle.dispatchOnStart()
+
+        assertEquals(AccountSyncState.Error, manager.accountSyncStateFlow.value)
+        verify { context.bindService(any(), any(), Context.BIND_AUTO_CREATE) }
+    }
+
+    @Test
+    fun `onStart when bindService succeeds state should be Loading before service calls back`() {
```

Comment:
```suggestion
    fun `onStart when Bitwarden app is present and bindService succeeds should be Loading before service calls back`() {
```

Reply:
I would argue that the app being present is part of `bindService` succeeding. But more detail is fine too:

b3dc685f283950ff2aa3026ba972a973f8ba4d14